search-improvements.rst

```diff
@@ -0,0 +1,189 @@
+
+  - Before: 746 ms
+  - After: 173 ms
+
+- `Project Doc Search`_: Search inside a project documentations from readthedocs.org domain.
+
+  - Before: 750 ms
+  - After: 270 ms
+
+
+Code Improvements
+^^^^^^^^^^^^^^^^^
+Code quality is very important in development world, specially in open source.
+As I have rewritten the search functionality from scratch, the code quality
+is improved in many ways like test coverage and documentation. So its easy for
+any contributor to start working on the search functionality
+
+Contributors Wanted
+^^^^^^^^^^^^^^^^^^^
+As Read The Docs is an open source project backed by a small team of developers,
+most of them are busy to keep things up and running only. Therefore, its quite
+hard for them to take time to implement new features. If you know some bit of
+Django or Python and Elasticsearch, you can contribute into the search functionality
+of Read The Docs. If you need any support to start contributing, you can get in touch with
+me or any member of  Read The Docs team. You can find all of us at `#readthedocs` freenode
+IRC channel or `readthedocs gitter`_ channel. I am `safwan` at IRC and `@safwanrahman`
+at gitter.
+
+Conclusion
+^^^^^^^^^^
+To conclude, I must say that the Search improvement in Read The Docs was very
+necessary and I'm glad I could improve it in such a short amount of time.
+There are an infinite number of ways it can be improved and I believe we can compete
+with major search engines in terms of documentation searching.
+Due to the constraints of only working for three months,
+a number of compelling features were left out such as `Search as You Type and Autocomplete`_ and
+`Code Search`_ functionality. Moreover, proper documentation is needed for the search
+architecture. I have tried to write test cases for most of the scenario, but because of
+time constrains, a lot of code is out of test coverage.
+
+
+I strongly hope that we will get the left behind work done within a short amount of time.
+This can be done easily if we get more contributors donate their time for improving Read The Docs.
+We don't need superhero or coding guru, just need people who understand Python, Django and
+Elasticsearch and have some time to write some code for us. You are a **Superhero** to us
+if you can lend your time and effort to improve Read The Docs.
+
+.. _Rob Hudson: https://github.com/robhudson
+.. _back in 2013: https://github.com/rtfd/readthedocs.org/pull/493
+.. _Elasticsearch: https://www.elastic.co/products/elasticsearch
+.. _Elasticsearch 1.3.x: https://www.elastic.co/guide/en/elasticsearch/reference/1.3/index.html
+.. _Elasticsearch 5.x: https://www.elastic.co/guide/en/elasticsearch/reference/5.4/index.html
+.. _Elasticsearch 6.x: https://www.elastic.co/guide/en/elasticsearch/reference/6.3/index.html
+.. _Elasticsearch 6.x has major changes: https://www.elastic.co/guide/en/elasticsearch/reference/current/release-notes-6.0.0.html
+.. _Project Ideas: https://git.io/fN9GK
```

Comment:
Why is this a short url?

Reply:
Because the URL is so big as I have taken it from github commit tag!
so prefered to use github short url